### PR TITLE
[G-API] Extend compileStreaming to support different overloads

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Overview
The goal of this PR is to support all different overloads for `compileStreaming` method in python, such as:
* compileStreaming(cv::GMetaArgs, cv::GCompileArgs = { })
* compileStreaming(cv::ExtractMetaCallback, cv::GCompileArgs = { })
* compileStreaming(cv::GCompileArgs = { })

To wrap these methods need to add `GAPI_WRAP` macros before every signature. The generated code will try every overloads in order from top to bottom as declared in c++ file and if overload isn't suitable (`pyopencv_to` for params return false) try the next one. So the problem was that `pyopencv_to(cv::ExtractMetaCallback)` always return `true` which means  that the `compileStreaming(cv::ExtractMetaCallback, cv::GCompileArgs = { })` is always suitable.
So the possible solution for that problem is put this overload in the end in that case it will be chosen only if others aren't suitable.

* Added `queue_capacity` compile arg in order to test overloads.
* Implemented `pyopencv_to(cv::GMetaArg)` in order to wrap `compileStreaming(cv::GMetaArgs, cv::GCompileArgs)`

### Build configuration
```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

Xbuild_image:Custom=centos:7
Xbuildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_image:Custom=ubuntu-openvino-2021.3.0:20.04
build_image:Custom Win=openvino-2021.2.0
build_image:Custom Mac=openvino-2021.2.0

test_modules:Custom=gapi,python2,python3,java
test_modules:Custom Win=gapi,python2,python3,java
test_modules:Custom Mac=gapi,python2,python3,java

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

File: modules/gapi/misc/python/pyopencv_gapi.hpp [hunk 1]
```diff
@@ -180,6 +181,42 @@ bool pyopencv_to(PyObject* obj, cv::gapi::wip::draw::Prim& value, const ArgInfo&
...
+        value = reinterpret_cast<pyopencv_GArrayDesc_t*>(obj)->v;
+    }
+    else if (PyObject_TypeCheck(obj,
+                reinterpret_cast<PyTypeObject*>(pyopencv_GOpaqueDesc_TypePtr)))
+    {
+        value = reinterpret_cast<pyopencv_GOpaqueDesc_t*>(obj)->v;
+    }
+    else
+    {
+        return false;
```
Comment (TolyaTalamanov) [inline modules/gapi/misc/python/pyopencv_gapi.hpp:209]: Need to set error

**Comment by TolyaTalamanov** [comment]
@mpashchenkov Could you have a look, please ?

File: modules/gapi/misc/python/pyopencv_gapi.hpp (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/pyopencv_gapi.hpp:210]: Done

File: modules/gapi/include/opencv2/gapi/gcomputation.hpp [hunk 2]
```diff
@@ -437,11 +437,7 @@ class GAPI_EXPORTS_W GComputation
      *
      * @sa @ref gapi_compile_args
      */
-    GStreamingCompiled compileStreaming(GMetaArgs &&in_metas, GCompileArgs &&args = {});
-
-    /// @private -- Exclude this function from OpenCV documentation
```
Comment (TolyaTalamanov) [inline modules/gapi/include/opencv2/gapi/gcomputation.hpp:442]: Return back comment

File: modules/gapi/misc/python/pyopencv_gapi.hpp [hunk 3]
```diff
@@ -180,6 +181,43 @@ bool pyopencv_to(PyObject* obj, cv::gapi::wip::draw::Prim& value, const ArgInfo&
...
+    }
+    return true;
+}
+
+template <>
+bool pyopencv_to(PyObject* obj, cv::GMetaArgs& value, const ArgInfo& info)
+{
+    return pyopencv_to_generic_vec(obj, value, info);
+}
```
Comment (OrestChura) [inline modules/gapi/misc/python/pyopencv_gapi.hpp:219]: I suggest to put all of it after `pyopencv_to(obj, cv::gapi::wip::draw::Prims&)` (the next overload) not to separate `Prim` and `Prims` overloads

**Review by dmatveev — APPROVED** [review]
LGTM but please address comments by @OrestChura

File: modules/gapi/misc/python/pyopencv_gapi.hpp [hunk 4]
```diff
@@ -180,6 +181,43 @@ bool pyopencv_to(PyObject* obj, cv::gapi::wip::draw::Prim& value, const ArgInfo&
...
+    {
+        value = reinterpret_cast<pyopencv_GOpaqueDesc_t*>(obj)->v;
+    }
+    else
+    {
+        failmsg("Unsupported cv::GMetaArg type");
+        return false;
+    }
+    return true;
+}
```
[the remark this answers is not in the document]
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/pyopencv_gapi.hpp:189]: Done

File: modules/gapi/misc/python/pyopencv_gapi.hpp (hunk 3 above)
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/pyopencv_gapi.hpp:210]: Done

File: modules/gapi/include/opencv2/gapi/gcomputation.hpp [hunk 5]
```diff
@@ -464,6 +460,10 @@ class GAPI_EXPORTS_W GComputation
      */
     GAPI_WRAP GStreamingCompiled compileStreaming(GCompileArgs &&args = {});
 
+    /// @private -- Exclude this function from OpenCV documentation
+    GAPI_WRAP GStreamingCompiled compileStreaming(const cv::detail::ExtractMetaCallback &callback,
+                                                  GCompileArgs                   &&args = {});
```
[the remark this answers is not in the document]
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/gcomputation.hpp]: Done

File: modules/gapi/include/opencv2/gapi/gstreaming.hpp [hunk 6]
```diff
@@ -396,10 +396,10 @@ namespace streaming {
  * In the streaming mode the pipeline steps are connected with queues
  * and this compile argument controls every queue's size.
  */
-struct GAPI_EXPORTS queue_capacity
+struct GAPI_EXPORTS_W_SIMPLE queue_capacity
 {
-    explicit queue_capacity(size_t cap = 1) : capacity(cap) { };
-    size_t capacity;
+    GAPI_WRAP explicit queue_capacity(size_t cap = 1) : capacity(cap) { };
+    GAPI_PROP size_t capacity;
```
[the remark this answers is not in the document]
Comment (TolyaTalamanov) [reply modules/gapi/include/opencv2/gapi/gstreaming.hpp]: Done

File: modules/gapi/misc/python/pyopencv_gapi.hpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/pyopencv_gapi.hpp:189]: Done

File: modules/gapi/misc/python/pyopencv_gapi.hpp (hunk 3 above)
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/pyopencv_gapi.hpp:210]: Done

**Pull request merged**